core/src/main/java/com/linecorp/armeria/client/endpoint/dns/DnsEndpointGroupBuilder.java

```diff
@@ -67,11 +67,15 @@ final String hostname() {
         return hostname;
     }
 
+    /**
+     * Returns the {@link EventLoop} set via {@link #eventLoop(EventLoop)} or acquires a random
+     * {@link EventLoop} from {@link CommonPools#workerGroup()}.
+     */
     final EventLoop eventLoop() {
         if (eventLoop != null) {
             return eventLoop;
         } else {
-            return CommonPools.workerGroup().next();
+            return eventLoop = CommonPools.workerGroup().next();
```

Comment:
Never expected that two different `EventLoops`s can be used. 😱 
Anyway, how about reverting this and calling `eventLoop()` before calling `buildResolver()`?
We can pass the `eventLoop` as a parameter of `buildResolver` methods. This will prevent setting a field from the getter method.

Reply:
Good idea. `eventLoop()` looks a getter method that returns the same value. Let me change the name to `getOrAcquireEventLoop()` to deliver the behavior correctly.